lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1331,8 +1357,670 @@
 
     // Intl.DateTimeFormat, Date.prototype.toLocaleString, Date.prototype.toLocaleDateString, Date.prototype.toLocaleTimeString
     var DateTimeFormat = (function () {
+        if ((InitType === 'Intl' || InitType === 'Date') && isPlatformUsingICU) {
+            const narrowShortLong = ["narrow", "short", "long"];
+            const twoDigitNumeric = ["2-digit", "numeric"];
+            const allOptionValues = [...twoDigitNumeric, ...narrowShortLong]
```

Comment:
_.concat is probably better for the same reason platform.builtInJavascriptArrayEntryConcat (which _.concat is implemented in terms of) -- tainting. I doubt there's a performance difference because platform.builtInJavascriptArrayEntryConcat is [essentially or actually] the same function that backs Array.prototype.concat (so you could do Array.prototype.concat.call(array, els) or array.concat(els) and it's the same. Taining of Array.prototype is the problem we're trying to avoid here because the actual function assigned to Array.prototype.concat is user-controllable

---
In reply to: [155607571](https://github.com/Microsoft/ChakraCore/pull/4390#discussion_r155607571) [](ancestors = 155607571)

Reply:
On that note we should ensure the tainting tests are updated to include any new platform functions you've exposed here, @jackhorton

---
In reply to: [155905000](https://github.com/Microsoft/ChakraCore/pull/4390#discussion_r155905000) [](ancestors = 155905000,155607571)